db/template.go

```diff
@@ -19,11 +19,17 @@ func (d TinkDB) CreateTemplate(ctx context.Context, name string, data string, id
 		return err
 	}
 
+	fields := map[string]string{
+		"name": name,
+	}
+	_, _, _, err = d.GetTemplate(ctx, fields, false)
```

Comment:
This query is not in the same transaction as the following insert DML, thus is not concurrent safe.